src/Symfony/Component/HttpClient/CurlHttpClient.php

```diff
@@ -138,12 +138,12 @@ public function request(string $method, string $url, array $options = []): Respo
             CURLOPT_CERTINFO => $options['capture_peer_cert_chain'],
         ];
 
-        if (\defined('CURL_VERSION_HTTP2') && (CURL_VERSION_HTTP2 & self::$curlVersion['features']) && ('https:' === $scheme || 2.0 === (float) $options['http_version'])) {
+        if (\defined('CURL_VERSION_HTTP2') && (CURL_VERSION_HTTP2 & self::$curlVersion['features']) && (('https:' === $scheme && null === $options['http_version']) || 2.0 === (float) $options['http_version'])) {
             $curlopts[CURLOPT_HTTP_VERSION] = CURL_HTTP_VERSION_2_0;
+        } elseif (('http:' === $scheme && null === $options['http_version']) || 1.1 === (float) $options['http_version']) {
```

Comment:
@nicolas-grekas I also restored the past behavior that set the http version to 1.1 for http request that have not specified the http_version option (see https://github.com/symfony/symfony/pull/36422/commits/a5b884cd94e792e7a57c99f3bebf69a4f1848697) - is that important?

Reply:
https: should always try http/2 first when no version is given